dask/array/core.py

```diff
@@ -2762,6 +2797,11 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     for i, ind in enumerate(inds):
         ind[axis] = -(i + 1)
 
+    metas = [getattr(s, '_meta', s) for s in seq]
+    metas = [m[tuple(slice(0, 0, None) for _ in range(m.ndim)) if
+             hasattr(m, 'ndim') else slice(0, 0, None)] for m in metas]
```

Comment:
You've mentioned this slicing before.  What is the purpose? Do we expect the metas to have correct dimensionality?  (this would maybe be ideal, but I suspect that it's quite hard to achieve)

Reply:
NM on this.  As long as we normalize it's probably easy to guarantee.

Reply:
Exactly, I've been trying to normalize dimensions, it's slightly more complex code-wise, but it gives us the ability to apply functions that rely heavily on the dimensionality, which is of utmost importance.